llm_ollama/__init__.py

```diff
@@ -14,6 +14,79 @@
 cache = Cache(llm.user_dir() / "llm-ollama" / "cache")
 
 
+@llm.hookimpl
+def register_tools(register):
+    """Register Ollama web search tools so LLM can execute them."""
+    # Set API key from LLM's key store if not already in environment
+    # This must happen before creating the Ollama client
+    import os
+    if not os.environ.get("OLLAMA_API_KEY"):
+        try:
+            key = llm.get_key("", "ollama", "OLLAMA_API_KEY")
+            if key:
+                os.environ["OLLAMA_API_KEY"] = key
+        except Exception:
+            pass  # Key not found, will rely on environment variable
+
+    try:
+        from ollama import Client
+    except ImportError:
+        return
+
+    # Create a client instance that will use the API key from environment
+    client = Client()
```

Comment:
Can we handle client in the same way as chat hooks? That is, a) instantiate just-in-time and b) use helpers from auth module?